idd/CMakeLists.txt

```diff
@@ -63,3 +63,27 @@ set(PREVIOUS_IDD "${CMAKE_CURRENT_SOURCE_DIR}/versions/V${PREV_RELEASE_MAJOR}-${
 set(PREVIOUS_IDD ${PREVIOUS_IDD} PARENT_SCOPE)
 
 install(FILES "${PREVIOUS_IDD}" DESTINATION "PreProcess/IDFVersionUpdater")
+
+if (BUILD_TESTING)
+  execute_process(COMMAND ${Python_EXECUTABLE} -m pytest --version
+    RESULT_VARIABLE _Pytest_STATUS
+        OUTPUT_VARIABLE Pytest_Version
+        ERROR_QUIET
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  if(_Pytest_STATUS AND NOT _Pytest_STATUS EQUAL 0)
+    message(AUTHOR_WARNING "Pytest isn't installed on your system python, so some tests won't be run. Run `${Python_EXECUTABLE} -m pip install pytest`")
+    set(Pytest_AVAILABLE OFF)
+  else()
+    message(VERBOSE "Found Pytest: ${Pytest_Version}")
+    set(Pytest_AVAILABLE ON)
+  endif()
+
+  if (Pytest_AVAILABLE)
+    add_test(NAME idd_schema.test_idd_parser
+      COMMAND ${Python_EXECUTABLE} -m pytest --verbose "${CMAKE_CURRENT_SOURCE_DIR}/schema/test_idd_parser.py"
+    )
+    set_tests_properties(idd_schema.test_idd_parser PROPERTIES TIMEOUT 30)
+  endif()
+
+endif()
```

Comment:
Register the pytest file as a ctest only if pytest is found. You'll get an author warning otherwise. I could lower the log level there, to VERBOSE maybe? @Myoldmopar thoughts?

I'm using a global timeout on the ctest. I originally had it implemented via the `pytest-timeout` plugin so individual pytests had a proper timeout, but I was wary of being annoying to developers by asking them to install two things.

Reply:
Pytest should be standardized if you ask me. it's much better than unittest. I went with it because of that, and because I was actively using pytest-timeout originally because it WAS hanging indefinitely unless I got the missing semi colon exception.